### PR TITLE
2537 Salinskii Alexander

### PR DESCRIPTION
**Pull request by asalynski**

### CODE REVIEW

**Comment by thevery** [comment]
refresh in action bar is nice

**Comment by thevery** [comment]
top/bottom padding is missing

**Comment by thevery** [comment]
action bar is useless on the second screen

**Comment by thevery** [comment]
first scrolling seems to be slow

**Comment by thevery** [comment]
app is ok (20-2=18), code review pending

**Comment by asalynski** [comment]
Sorry, but in the def of the problem you didn't said anything about top/bottom padding.

**Comment by thevery** [comment]
Yes. That's what we call "common sense" in our tasks' special condition "Ничего из условия выше не заменит здравого смысла."

File: AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          package="com.example.extratask1"
```
Comment (thevery) [inline AndroidManifest.xml:3]: bad package

File: res/layout/pictures.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (thevery) [inline res/layout/pictures.xml:3]: useless layout

File: src/com/example/extratask1/DataBases.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        onCreate(database);
+    }
+
+    public long newPicture(byte[] picture)
+    {
+        SQLiteDatabase database = this.getWritableDatabase();
+        ContentValues c_values = new ContentValues();
+        long id;
+        Date date = new Date();
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault());
```
Comment (thevery) [inline src/com/example/extratask1/DataBases.java:60]: locale-dependent format might change with locale change

File: src/com/example/extratask1/DataBases.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            do
+            {
+                byte[] bytes = cursor.getBlob(cursor.getColumnIndex(KEY_PICTURE));
+                MyPictures picture = new MyPictures(bytes);
+                picture.newID(cursor.getInt((cursor.getColumnIndex(KEY_ID))));
+                pictures.add(picture);
+            } while (cursor.moveToNext());
+        }
+
+        cursor.close();
```
Comment (thevery) [inline src/com/example/extratask1/DataBases.java:89]: use try/finally

File: src/com/example/extratask1/DownloadPictures.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        {
+
+            URL pictureUrl = new URL(url);
+            URLConnection ucon = pictureUrl.openConnection();
+            InputStream iStream = ucon.getInputStream();
+            BufferedInputStream buffIStream = new BufferedInputStream(iStream);
+            ByteArrayBuffer byteArrBuffer = new ByteArrayBuffer(700);
+            int current = 0;
+
+            while ((current = buffIStream.read()) != -1) {
```
Comment (thevery) [inline src/com/example/extratask1/DownloadPictures.java:93]: reading by 1 byte is a bad solution

File: src/com/example/extratask1/DownloadPictures.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+                byteArrBuffer.append((byte) current);
+            }
+
+            return byteArrBuffer.toByteArray();
+
+        } catch (Exception e) {}
```
Comment (thevery) [inline src/com/example/extratask1/DownloadPictures.java:99]: why do you ignore exception here?

File: src/com/example/extratask1/FullscreenPictureActivity.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+{
+
+    public void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.fsp);
+
+        DataBases database = new DataBases(getApplicationContext());
+        ImageView imageView = (ImageView) findViewById(R.id.imageView);
+        int id = getIntent().getIntExtra("id", 0);
```
Comment (thevery) [inline src/com/example/extratask1/FullscreenPictureActivity.java:24]: `"id"` should be const

File: src/com/example/extratask1/MyActivity.java [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+                item.setEnabled(false);
+                downloadPictures();
+                return true;
+        }
+
+        return super.onMenuItemSelected(featureId, item);
+    }
+
+    private void downloadPictures() {
+        Toast toast = Toast.makeText(getApplicationContext(),"Downloading...", Toast.LENGTH_SHORT);
```
Comment (thevery) [inline src/com/example/extratask1/MyActivity.java:84]: strings.xml

File: src/com/example/extratask1/MyActivity.java [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    private void viewConfig(GridView tmpGridView) {
+
+        gridView = tmpGridView;
+        Point display_size = new Point();
+        getWindowManager().getDefaultDisplay().getSize(display_size);
+        int width = display_size.x;
+
+        if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT)
```
Comment (thevery) [inline src/com/example/extratask1/MyActivity.java:98]: use `layout` (or `values`) and `layout-land` (`values-land`) resource folders instead

File: src/com/example/extratask1/URLParser.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        try
+        {
+
+            JSONArray array = jsonObject.getJSONArray("entries");
+
+            for (int i = 0; i < array.length(); i++)
+            {
+
+                JSONObject photo = array.getJSONObject(i);
+                JSONObject img = photo.getJSONObject("img");
```
Comment (thevery) [inline src/com/example/extratask1/URLParser.java:34]: const here as well

**Comment by thevery** [comment]
20-2-4=14

**Pull request closed without merging**